src/rez/tests/test_shells.py

```diff
@@ -189,21 +231,29 @@ def _execute_code(func, expected_output):
 
             out, _ = p.communicate()
             self.assertEqual(p.returncode, 0)
-            token = '\r\n' if platform_.name == 'windows' else '\n'
+
+            # Powershell and Unix uses \n
+            sh = create_shell()
+            token = '\r\n' if sh.name() == 'cmd' else '\n'
+
             output = out.strip().split(token)
             self.assertEqual(output, expected_output)
 
         def _rex_assigning():
-            import os
-            windows = os.name == "nt"
+            from rez.shells import create_shell, UnixShell
+            sh = create_shell()
+            is_powershell = sh.name() in ["powershell", "pwsh"]
 
             def _print(value):
                 env.FOO = value
                 # Wrap the output in quotes to prevent the shell from
                 # interpreting parts of our output as commands. This can happen
                 # when we include special characters (&, <, >, ^) in a
                 # variable.
-                info('"%FOO%"' if windows else '"${FOO}"')
+                if is_powershell:
+                    info('`"{}`"'.format(sh.get_key_token("FOO")))
+                else:
+                    info('"{}"'.format(sh.get_key_token("FOO")))
```

Comment:
So the original code here should really just have been:

    info('"${FOO}"')

The test is showing two things - that ${X}/$X variable references are getting correctly translated to their shell equivalent; and that use of expandable/literal is working as expected. Rather than directly use `get_key_token` anywhere here, the idea was just to test the actual output from the shell.

Wrt the whole adding of double quotes here to avoid problems with '>' etc, it does seem like that should be dealt with by `info` itself (and by shell-specific string expansion in general). But that stuff gets tricky and a change related to that should be in a separate PR. Is Powershell a special case where this has to be done differently?

Reply:
This now passes and I was able to do the right thing by introducing `convert_tokens` to Shell.
This was actually simpler that both of us thought 👍